examples/opengl/cube/source/app.d

```diff
@@ -0,0 +1,533 @@
+module app;
+
+import vecs;
+
+import bindbc.glfw.bindstatic;
+import bindbc.glfw.types;
+import bindbc.opengl;
+import dlib.geometry;
+import dlib.math;
+
+import std.experimental.logger : errorf, trace, tracef;
+import std.string : format, fromStringz, toStringz;
+import std.typecons : Tuple, tuple;
+
+
+static immutable vertexShader = q{
+#version 330 core
+
+layout(location = 0) in vec3 pos;
+uniform mat4 view, proj, model;
+
+out vec3 pos_color;
+
+void main() {
+	gl_Position =  proj * view * model * vec4(pos.xyz, 1.0f);
+	pos_color = vec3(clamp(pos.x, 0.1, 1), clamp(pos.y, 0.1, 1), clamp(pos.z, 0.1, 1));
+}};
+
+
+static immutable fragmentShader = q{
+#version 330 core
+
+in vec3 pos_color;
+out vec4 Color;
+
+void main() {
+    Color = vec4(pos_color, 1.0f);
+}};
+
+// super simplistic input manager
+bool[int] keys;
+
+
+  // ==============
+ // # Components #
+// ==============
+
+enum Input;
+enum Graphics;
+
+struct Transform
+{
+	vec3 position;
+	vec3 size;
+	vec3 rotation;
+}
+
+struct Model
+{
+	vec3 position;
+	vec3 scale;
+	vec3 rotation;
+}
+
+struct Camera
+{
+	vec3 position;
+	Tuple!(float, "fov", float, "aspectRatio", float, "near", float, "far") frustum;
```

Comment:
Shouldn't this be a struct, instead?

Reply:
Same thing but with more work. I would have to create a struct and define an instance of it. The correct way would be to define the variables without being attached to anything. But I wanted to ensure that those variables define the `frustum`.